### PR TITLE
Fix weight dtypes

### PR DESCRIPTION
**Pull request by gobbleturk**
Fix weight dytpes to float32

Weight dtypes inadvertently changed to bf16 in PR #417 

Ran convergence test and confirmed familiar good convergence of 2.48 after 20k steps [XPK logs](https://pantheon.corp.google.com/logs/query;query=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22tpu-prod-env-multipod%22%0Aresource.labels.location%3D%22us-central2%22%0Aresource.labels.cluster_name%3D%22v4-bodaborg%22%0Aresource.labels.namespace_name%3D%22default%22%0Alabels.k8s-pod%2Fjobset_sigs_k8s_io%2Fjobset-name%3D%22mattdavidow-fix-dtype-a2%22%20severity%3E%3DDEFAULT%0Aresource.labels.pod_name%3D%22mattdavidow-fix-dtype-a2-slice-job-0-1-k4j7z%22;cursorTimestamp=2024-03-02T08:32:05.916493434Z;startTime=2024-03-01T23:50:42.346Z;endTime=2024-03-02T20:50:42.346351Z?e=13803378&mods=allow_workbench_image_override&project=tpu-prod-env-multipod)

### CODE REVIEW

**Review by rwitten — APPROVED** [review]

File: MaxText/layers/normalizations.py [hunk 1]
```diff
@@ -42,7 +43,7 @@ def __call__(self, x: jnp.ndarray) -> jnp.ndarray:
         'scale',
         nn.with_logical_partitioning(self.scale_init, self.kernel_axes),
         (features,),
-        self.dtype,
+        self.weight_dtype,
     )
 
     scale = jnp.asarray(scale, self.dtype)
```
[the remark this answers is not in the document]
Comment (gobbleturk) [reply MaxText/layers/normalizations.py:49]: It changes the data type of the learned weights, but does not affect the computational data type. In your code snippet the scale is recast from` weight_dtype` to `dtype` before using in computation. Similarly for the kernel in linears.py - the dtype in kernel refers to the data type of learned weights (we set to `weight_dtype`), but then we recast to `dtype` before using computation

[the remark this answers is not in the document]
Comment (NinaCai) [reply MaxText/layers/normalizations.py:49]: I see. Makes perfect sense. Thanks

**Review by rwitten — APPROVED** [review]
